syntax = "proto3";

package users;

service UserService {
  rpc GetUsers(Empty) returns (UserList) {}
  rpc GetUserById(UserRequest) returns (User) {}
  rpc GetGPSData(Empty) returns (stream GPSData) {}
}

message Empty {}

message UserRequest {
  string userId = 1;
}

message User {
  string userId = 1;
  string name = 2;
  string email = 3;
}

message UserList {
  repeated User users = 1;
}

message GPSData {
    string userId = 1;
    float latitude = 2;
    float longitude= 3;
}